lighthouse-core/gather/drivers/driver.js

```diff
@@ -213,6 +221,75 @@ class Driver {
     });
   }
 
+  getServiceWorkerRegistrations() {
+    return new Promise(resolve => {
+      this.once('ServiceWorker.workerRegistrationUpdated', data => {
+        this.sendCommand('ServiceWorker.disable');
+        resolve(data);
+      });
+
+      this.sendCommand('ServiceWorker.enable');
+    });
+  }
+
+  /**
+   * Checks all service workers and see if any duplications are running
```

Comment:
something like `Rejects if any open tabs would share a service worker with the target URL.`?